core/src/main/scala/org/apache/spark/scheduler/DAGScheduler.scala

```diff
@@ -493,7 +495,8 @@ class DAGScheduler(
     val func2 = func.asInstanceOf[(TaskContext, Iterator[_]) => _]
     val waiter = new JobWaiter(this, jobId, partitions.size, resultHandler)
     eventProcessLoop.post(JobSubmitted(
-      jobId, rdd, func2, partitions.toArray, allowLocal, callSite, waiter, properties))
+      jobId, rdd, func2, partitions.toArray, allowLocal, callSite, waiter,
+      SerializationUtils.clone(properties)))
```

Comment:
We need to use `SerializationUtils.clone()` in order to perform a deep-clone which also captures properties that were inherited from parents; see 5a5526164bdf9ecf1306d4570e816eb4df5cfd2b for more discussion of this issue.